src/components/GlobalPreferences/Notifications/notification-service-api.js

```diff
@@ -46,22 +43,55 @@ export async function verifyEmailToken(shortLivedToken) {
       },
     })
 
-    if (!rawResponse.ok) {
-      // In case of errors the api will return json payload
-      const response = await rawResponse.json()
+    if (rawResponse.ok) {
+      return rawResponse.headers.get('authorization')
+    }
 
-      if (isTokenExpired(response))
-        throw new ExpiredTokenError(response.message)
+    // In case of errors the api will return json payload
+    const response = await rawResponse.json()
 
-      if (isUnauthorized(rawResponse)) {
-        throw new UnauthroizedError(rawResponse.statusText)
-      }
+    if (isTokenExpired(response)) throw new ExpiredTokenError(response.message)
 
-      throw new Error(rawResponse.statusText)
+    if (isUnauthorized(rawResponse)) {
+      throw new UnauthorizedError(rawResponse.statusText)
     }
 
+    throw new Error(rawResponse.statusText)
+
     // Get the long lived token from header
-    return rawResponse.headers.get('authorization')
+  } catch (e) {
+    console.error(e.message)
+    throw e
+  }
+}
+
+// Verify that the long lived token is valid and has not expired
+export async function isTokenValid(longLivedToken) {
```

Comment:
Had a thought, what do you think about renaming `token` to `authToken` or similar in this file?

Just to avoid any potential confusion with ERC20 tokens.

Reply:
Absolutely